src/movies.js

```diff
@@ -1,22 +1,64 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    let movie = moviesArray.map ((movie)=> movie.director)
+    return movie
+};
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
-
+function howManyMovies(moviesArray) {
+    if (moviesArray === 0) {
+            return 0
+        }
+        const moviesSpeilberg = moviesArray.filter(allSpielberg => {
+            return allSpielberg.director === "Steven Spielberg" && allSpielberg.genre.includes("Drama")
+        })
+    return moviesSpeilberg.length    
+};
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
```

Comment:
This is correct and it's working, but on the tests they ask you to also `return 0` if the `array` is empty, and that the function should be able to work even if one of the movies doesn't have the `key` `score`.

So for this you need to add two `conditions`, one to check the `length` on the array and `return 0` if it is empty:
```
  if (!moviesArray.length) return 0;
```

And another one to check if the `elements` of the `array` have the `key` `score` if it doesn't continue counting the `score` of the other `elements`:
```
    if (!vall.score) {
      return acc;
    }
```

So at the end this function should look something like this:
```
function scoresAverage(moviesArray) {
  if (!moviesArray.length) return 0;

  const average = moviesArray.reduce((acc, vall) => {
    if (!vall.score) {
      return acc;
    }

    return acc + vall.score;
  }, 0);
  const averageTotal = average / moviesArray.length;
  return +averageTotal.toFixed(2);
}
```